Week3/index.js

```diff
@@ -0,0 +1,77 @@
+const mysql = require('mysql');
+const express = require('express');
+const session = require('express-session');
+const bodyParser = require('body-parser');
+const path = require('path');
+const sqlconnection = require('./sqlqueries.js');
+const util = require('util');
+const pug = require('pug');
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'todoapp',
+});
+
+const app = express();
+
+const execQuery = util.promisify(connection.query.bind(connection));
+
+app.use(
+  session({
+    secret: 'elpsykongroo',
+    resave: true,
+    saveUninitialized: true,
+  }),
+);
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+app.set('view engine', 'pug');
+
+app.get('/', function(request, response) {
+  response.sendFile(path.join(__dirname + '/index.html'));
+});
+
+app.post('/auth', function(request, response) {
+  const username = request.body.username;
+  const password = request.body.password;
+  if (username && password) {
+    connection.query(
+      'SELECT * FROM users WHERE username = ? AND password = ?',
+      [username, password],
+      function(error, results, fields) {
+        if (results.length > 0) {

```

Comment:
Check the error first here, if an error is raised result will be undefined.

Reply:
I actually had it right there but removed it to simplify the code a bit, will add it back again :)